### PR TITLE
Fix sloppy code mixing `int` and `size_t` after `cc1` no more supporting `-Wno-shorten-64-to-32`

### PR DESCRIPTION
**Pull request by DDvO**
This is ~not yet~ meanwhile complete.
We meanwhile also added other related fixes to `Makefile_v1` and `CMakeLists.txt`
and added a couple of comments to `config_update.c`.

### CODE REVIEW

**Review by DDvO — COMMENTED** [review]
A couple of comments

File: Makefile_v1 [hunk 1]
```diff
@@ -103,7 +108,7 @@ DEST_INC=$(DEST_PRE)
 DEST_DOC=$(DEST_PRE)/share/doc/libsecutils-dev
 OUTBIN=icvutil$(EXE)
 DEST_BIN=$(DEST_PRE)/bin
-LOCAL_CFLAGS=-std=gnu90 -fPIC
```
[the remark this answers is not in the document]
Comment (kiron-mx) [reply Makefile_v1:106]: Enabling `std=gnu90` leads to (amongst others) a lot of warnings concerning the ISO C90 standard (`-Werror=pedantic`)

[the remark this answers is not in the document]
Comment (DDvO) [reply Makefile_v1:106]: I see- then please put it back in commented form, marking it as 'TODO' at the end of the line,
and do so also for `CMakeLists.txt`.

File: src/config/config_update.c [hunk 2]
```diff
@@ -387,7 +391,7 @@ int CONF_update_config(OPTIONAL ossl_unused uta_ctx* ctx, const char* file_name,
             return 0;
         }
 #endif
-        return file_len;
+        return (int)file_len;
```
[the remark this answers is not in the document]
Comment (kiron-mx) [reply src/config/config_update.c:394]: This also apply to the cast in line 400, right?

[the remark this answers is not in the document]
Comment (DDvO) [reply src/config/config_update.c:394]: Yes. Good that you noticed that further occurrence.

File: Makefile_v1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (DDvO) [reply Makefile_v1:106]: I am still missing the equivalent on the `CMakeLists.txt` side.

File: src/config/config_update.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiron-mx) [reply src/config/config_update.c]: `file_len` is of type `size_t` but the return value of the function is `int`. Therefore, I kept the cast here. But as discussed, I added a TODO comment to check why the function always returns 0.

**Comment by DDvO** [comment]
@benjamin-schilling could you please have a look?

**Review by benjamin-schilling — APPROVED** [review]

**Pull request merged**